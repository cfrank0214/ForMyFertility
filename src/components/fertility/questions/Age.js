import React, { Component } from 'react';
import { ProgressViewIOS, Picker } from 'react-native';

import {
    Control,
    Form,
    actions,
} from 'react-redux-form/native';
import {
    Content,
    Text,
    View
} from 'native-base';

let ages = [];
for (let i = 2005; i >= 1955; i--) {
    ages.push(<Picker.Item key={i} label={`${i}`} value={i} />)
}

export default class Age extends Component {

    handleSubmit(values) {
        this.props.dispatch(actions.submit('fertilityQuestions', values));
    }
    render() {
        return (
            <Content >
                <View style={{ alignSelf: "center", margin: 10 }}>
                    <Text>Question 1 of 7</Text>
                    <ProgressViewIOS progress={0.125} progressTintColor={'#86B2CA'} />
                </View>
                <Form model="fertilityQuestions" onSubmit={values => this.handleSubmit(values)}>
                    <Text>What year were you born?</Text>
                    <Control.Picker model=".age">
                        {ages}
                    </Control.Picker>
                </Form >
            </Content>
        );
    }
}