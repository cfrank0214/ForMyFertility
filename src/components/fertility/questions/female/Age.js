import React from 'react';
import { connect } from 'react-redux';
import { Picker } from 'react-native';
import * as Progress from 'react-native-progress';
import {
    Control,
    Form,
    actions,
} from 'react-redux-form/native';
import {
    Content,
    Button,
    Text,
    View
} from 'native-base';

let ages = [];
let curYear = (new Date()).getFullYear();
let youngFertility = curYear - 13;
let oldFertility = curYear - 60;
let  questionNum, questionTot;
for (let i = youngFertility; i >= oldFertility; i--) {
    ages.push(<Picker.Item key={i} label={`${i}`} value={i} />)
}

export class Age extends React.Component {

    handleSubmit({ questionNum, dispatch, navigation }) {
        dispatch(actions.change('fertilityQuestions.questionNum', ++questionNum));
        navigation.navigate('YearTrying')
    }

    render() {
        questionNum = this.props.fertilityQuestions ? this.props.fertilityQuestions.questionNum : 1;
        questionTot = this.props.fertilityQuestions ? this.props.fertilityQuestions.questionTot : 1;;
        const { dispatch } = this.props;
        const navigation = this.props.navigation;
        return (
            <Content >
                <View style={{ alignSelf: "center", margin: 10 }}>
                    <Text> Questions {questionNum} of {questionTot}</Text>
                    <Progress.Bar progress={(questionNum / questionTot)} width={200} />
                </View>
                <Form model="fertilityQuestions" >
                    <Text>What year were you born?</Text>
                    <Control.Picker model=".age" >
                        {ages}
                    </Control.Picker>
                    <View>
                        <Button full rounded primary onPress={() => this.handleSubmit({ questionNum, dispatch, navigation })}>
                            <Text>Next</Text>
                        </Button>
                    </View>
                </Form>
            </Content>
        )
    }
}

function mapStateToProps(state) {
    return {
      nav: state.nav,
      fertilityQuestions: state.fertilityQuestions
    };
  }

export default connect(mapStateToProps)(Age);
