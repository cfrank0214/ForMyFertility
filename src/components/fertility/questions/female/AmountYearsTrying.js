import React from 'react';
import { connect } from 'react-redux';
import * as Progress from 'react-native-progress';
import { Picker } from 'react-native';
import {
    Control,
    Form,
    actions,
} from 'react-redux-form/native';
import {
    Content,
    Button,
    Text,
    View
} from 'native-base';

let years = [];
for (let i = 1; i < 20; i++) {
    years.push(<Picker.Item key={i} label={`${i}`} value={i} />)
}
function handleSubmit({ navigation, fertilityQuestions, dispatch, }) {
    dispatch(actions.change('fertilityQuestions.questionNum', ++fertilityQuestions.questionNum));
    navigation.navigate("Pregnant")
}

const AmountYearsTrying = ({ navigation, fertilityQuestions, dispatch }) => (
    <Content>
        <View style={{ alignSelf: "center", margin: 10 }}>
            <Text> Questions {fertilityQuestions.questionNum} of {fertilityQuestions.questionTot}</Text>
            <Progress.Bar progress={(fertilityQuestions.questionNum / fertilityQuestions.questionTot)} width={200} />
        </View>
        <Form model="fertilityQuestions" onSubmit={values => this.handleSubmit(values)}>
            <Text>How many years have you been trying to get pregnant?</Text>
            <Control.Picker model=".amountYearsChildlessSex">
                {years}
            </Control.Picker>
            <View>
                <Button full rounded primary onPress={() => handleSubmit({ navigation, fertilityQuestions, dispatch })}>
                    <Text>Next</Text>
                </Button>
            </View>
        </Form>
    </Content>
)
export default connect(({ nav, fertilityQuestions }) => ({ nav, fertilityQuestions }))(AmountYearsTrying);