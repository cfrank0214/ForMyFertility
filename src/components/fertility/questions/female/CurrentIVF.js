import React from 'react';
import { connect } from 'react-redux';
import {
    Control,
    Form,
    actions,
} from 'react-redux-form/native';
import {
    Picker,
} from 'react-native'
import * as Progress from 'react-native-progress';
import {
    Button,
    Content,
    Text,
    View
} from 'native-base';

function handleSubmit({ navigation, fertilityQuestions, dispatch, }) {
    dispatch(actions.change('fertilityQuestions.questionNum', ++fertilityQuestions.questionNum));
    if (fertilityQuestions.currentIVF === "yes") {
        dispatch(actions.change('fertilityQuestions.questionTot', (fertilityQuestions.questionTot+8)));
        navigation.navigate("Eggs")
    } else {
        navigation.navigate("Report")
    }
}

const CurrentIVF = ({ navigation, fertilityQuestions, dispatch }) => (
    <Content >
        <View style={{ alignSelf: "center", margin: 10 }}>
            <Text> Questions {fertilityQuestions.questionNum} of {fertilityQuestions.questionTot}</Text>
            <Progress.Bar progress={(fertilityQuestions.questionNum / fertilityQuestions.questionTot)} width={200} />
        </View>
        <Form model="fertilityQuestions" onSubmit={values => this.handleSubmit(values)}>
            <Text>Would you like to know your chances using IVF?</Text>
            <Control.Picker model=".currentIVF">
                <Picker.Item label='No' value='no' />
                <Picker.Item label='Yes' value='yes' />
            </Control.Picker>
            <View>
                <Button type="submit" full rounded primary onPress={() => handleSubmit({ navigation, fertilityQuestions, dispatch })}>
                    <Text>Next</Text>
                </Button>
            </View>
        </Form>
    </Content>
)

export default connect(({ nav, fertilityQuestions }) => ({ nav, fertilityQuestions }))(CurrentIVF);