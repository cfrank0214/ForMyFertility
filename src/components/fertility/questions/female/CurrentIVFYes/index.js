import Eggs from './Eggs';
import Embryos from './Embryos';
import EmbryosTransfer from './EmbryosTransfer';
import FrozenEmbryos from './FrozenEmbryos';
import Hormone from './Hormone';
import ICSI from './ICSI';
import IVFCYCLES from './IVFCYCLES';

export {
    Eggs, Embryos, EmbryosTransfer, FrozenEmbryos, Hormone, ICSI, IVFCYCLES
};