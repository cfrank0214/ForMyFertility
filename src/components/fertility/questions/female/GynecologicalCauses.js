import React from 'react';
import { connect } from 'react-redux';
import { Picker } from 'react-native';
import * as Progress from 'react-native-progress';
import {
    Control,
    Form,
    actions,
} from 'react-redux-form/native';
import {
    Button,
    Content,
    Text,
    View
} from 'native-base';

function handleSubmit({ navigation, fertilityQuestions, dispatch, }) {
    dispatch(actions.change('fertilityQuestions.questionNum', ++fertilityQuestions.questionNum));
    if (fertilityQuestions.gynecologicalCauses === "yes") {
        dispatch(actions.change('fertilityQuestions.questionTot', ++fertilityQuestions.questionTot));
        navigation.navigate("WhichGynecologicalCauses")
    } else {
        navigation.navigate("Partner")
    }
}

const GynecologicalCauses = ({ navigation, fertilityQuestions, dispatch }) => (
    <Content >
        <View style={{ alignSelf: "center", margin: 10 }}>
            <Text> Questions {fertilityQuestions.questionNum} of {fertilityQuestions.questionTot}</Text>
            <Progress.Bar progress={(fertilityQuestions.questionNum / fertilityQuestions.questionTot)} width={200} />
        </View>
        <Form model="fertilityQuestions" onSubmit={values => this.handleSubmit(values)}>
            <Text>Do you have any known gynecological or other causes of infertility?</Text>
            <Control.Picker model=".gynecologicalCauses" >
                <Picker.Item label='No' value='no' />
                <Picker.Item label='Yes' value='yes' />
            </Control.Picker>
            <View>
                <Button full rounded primary onPress={() => handleSubmit({ navigation, fertilityQuestions, dispatch })}>
                    <Text>Next</Text>
                </Button>
            </View>
        </Form>
    </Content>
)

export default connect(({ nav, fertilityQuestions }) => ({ nav, fertilityQuestions }))(GynecologicalCauses); 