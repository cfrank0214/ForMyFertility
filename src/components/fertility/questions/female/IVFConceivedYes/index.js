import AmountPerfectHealth from './AmountPerfectHealth';
import EmbryosFinalCycle from './EmbryosFinalCycle';
import MoreThanOneIVFConceived from './MoreThanOneIVFConceived';
import SingletonMultiBirth from './SingletonMultiBirth';

export {
    AmountPerfectHealth, EmbryosFinalCycle, MoreThanOneIVFConceived, SingletonMultiBirth
};