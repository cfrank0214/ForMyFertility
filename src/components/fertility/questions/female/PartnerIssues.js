import React from 'react';
import { connect } from 'react-redux';
import * as Progress from 'react-native-progress';
import { Picker } from 'react-native';
import {
    Control,
    Form,
    actions,
} from 'react-redux-form/native';
import {
    Button,
    Content,
    Text,
    View
} from 'native-base';

function handleSubmit({ navigation, fertilityQuestions, dispatch, }) {
    dispatch(actions.change('fertilityQuestions.questionNum', ++fertilityQuestions.questionNum));
    navigation.navigate("CurrentIVF")
}

const PartnarIssues = ({ navigation, fertilityQuestions, dispatch }) => (
    <Content >
        <View style={{ alignSelf: "center", margin: 10 }}>
            <Text> Questions {fertilityQuestions.questionNum} of {fertilityQuestions.questionTot}</Text>
            <Progress.Bar progress={(fertilityQuestions.questionNum / fertilityQuestions.questionTot)} width={200} />
        </View>
        <Form model="fertilityQuestions" onSubmit={values => this.handleSubmit(values)}>
            <Text>Which of the following describes his condition most-appropriately?</Text>
            <Control.Picker model=".whichPartnerIssue">
                <Picker.Item label='Azoospermia' value='azoospermia' />
                <Picker.Item label='Tubal Factor' value='tubalfactor' />
                <Picker.Item label='Idiopathy' value='idiopathy' />
                <Picker.Item label='Low Sperm Count' value='oligospermia' />
                <Picker.Item label='Poor Sperm Quality' value='poorspermquality' />
                <Picker.Item label='Poor Sperm Motility' value='poorspermmotility' />
            </Control.Picker>
            <View>
                <Button full rounded primary onPress={() => handleSubmit({ navigation, fertilityQuestions, dispatch })}>
                    <Text>Next</Text>
                </Button>
            </View>
        </Form>
    </Content>
);
export default connect(({ nav, fertilityQuestions }) => ({ nav, fertilityQuestions }))(PartnarIssues); 