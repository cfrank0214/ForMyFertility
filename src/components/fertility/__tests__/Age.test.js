/**
 * @jest-environment jsdom
 */
import React from 'react'
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { Age } from '../questions/female/Age';
import { Button, View } from 'native-base';
import * as Progress from 'react-native-progress';
import { Form } from 'react-redux-form/native';

describe('>>>AGE --- Shallow Render REACT COMPONENTS',()=>{
  let wrapper
  
  beforeEach(()=>{
      wrapper = shallow(<Age />)
      
  })

  it('renders correctly', () => {
    const tree = renderer.create(
      <Age />
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('+++ render the DUMB component', () => {
     expect(wrapper.length).toEqual(1)
  });
  it('contains 2 <View /> components', () => {
    expect(wrapper.find(View).length).toBe(2);
  });
  it('contains 1 <Form /> component', () => {
    expect(wrapper.find(Form).length).toBe(1);
  });
  it('contains 1 <Button /> component', () => {
    expect(wrapper.find(Button).length).toBe(1);
  });  
  it('contains 1 <Progress.Bar /> component', () => {
    expect(wrapper.find(Progress.Bar).length).toBe(1);
  }); 
});