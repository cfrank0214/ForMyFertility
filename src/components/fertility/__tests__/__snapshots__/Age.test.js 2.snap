// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>AGE --- Shallow Render REACT COMPONENTS renders correctly 1`] = `null`;
